include/storage/io.hpp

```diff
@@ -119,10 +120,30 @@ class FileReader
     {
         auto fingerprint = ReadOne<util::FingerPrint>();
         const auto valid = util::FingerPrint::GetValid();
-        // compare the compilation state stored in the fingerprint
-        return valid.IsMagicNumberOK(fingerprint) && valid.TestContractor(fingerprint) &&
-               valid.TestGraphUtil(fingerprint) && valid.TestRTree(fingerprint) &&
-               valid.TestQueryObjects(fingerprint);
+
+        // Check magic number
+        if (!valid.IsMagicNumberSame(fingerprint))
+        {
+            util::Log(logERROR) << "Fingerprint magic number does not match in "
+                                << filepath.string();
+            return false;
+        }
+
+        // This check also validates the major version
```

Comment:
Yeah, you're right.  I've dropped this approach, there is now just a single `IsDataCompatible()` function, which hopefully has a clearer intent.